drivers/timer/arcv2_timer0.c

```diff
@@ -229,20 +290,28 @@ u32_t z_clock_elapsed(void)
 		return 0;
 	}
 
+#ifdef CONFIG_SMP
+	return (z_arc_connect_gfrc_read() - last_time) / CYC_PER_TICK;
+#else
 	k_spinlock_key_t key = k_spin_lock(&lock);
 	u32_t cyc = elapsed();
 
 	k_spin_unlock(&lock, key);
 	return cyc / CYC_PER_TICK;
+#endif
 }
 
 u32_t z_timer_cycle_get_32(void)
 {
+#ifdef CONFIG_SMP
+	return z_arc_connect_gfrc_read() - start_time;
```

Comment:
i have considered to reset the gfrc,  but in hsdk, the reset/halt of gfrc don't work, i.e, gfrc cannot be controlled, so i introdcued start_time. Because start_time is  just set once, no need of a spinlock

Reply:
My point was: do we care about the start_time, or is it ok to return the slightly larger hw timer value directly?

Reply:
yes, e.g, soft reset, debug,  the gfrc will not stop, if no start_time,  when z_timer_cycl_get_32 first,called, it may return a very big value, like 0xa5a5a5a5, it will be unreasonable,  the whole system will work uncorrectly. If we can reset and clear the gfrc during init, the start_time can be removed.